R/lint.R

```diff
@@ -128,7 +128,8 @@ lint <- function(filename, linters = NULL, ..., cache = FALSE, parse_settings =
 lint_dir <- function(path = ".", ...,
                      relative_path = TRUE,
                      exclusions = list("renv", "packrat"),
-                     pattern = rex(".", one_of("Rr"), or("", "html", "md", "nw", "rst", "tex", "txt"), end),
+                     pattern = rex(".", or(group(one_of("Rr"), or("", "html", "md", "nw", "rst", "tex", "txt")),
```

Comment:
No need for a capturing `group()`, `list()` should work just fine IINM.

Reply:
I've just tested this: using `list()` instead of `group()' doesn't work - no files are linted in tests.

Reply:
@AshesITR I was also skeptical, but note that `rex:::group()` is a _non-capturing_ group:

```r
rex(group(or("a", "bc")))
# (?:(?:a|bc))
```

Reply:
Thanks for double checking and sorry for the confusion.